emcc.py

```diff
@@ -2016,7 +2016,11 @@ def phase_linker_setup(options, state, newargs, user_settings):
         '_wasmfs_chdir',
         '_wasmfs_symlink',
         '_wasmfs_chmod',
+        '_wasmfs_identify',
       ]
+    if settings.MAIN_MODULE:
+      # Dynamic library support uses JS API internals, so include it all
```

Comment:
Where is this usage exactly?  Can this be limited to `MAIN_MODULE=1`?

Reply:
Dynamic linking looks like it has just one call to `FS.findObject`.. should we wrap that in `#if FILESYSTEM` instead of this?

Reply:
I think it's not optional, though, is it? Dynamic linking needs filesystem support to be able to read library files.

Reply:
> No, it can fall back using readAsync to load directly without FS support.

readAsync can't help `dlopen()`, can it? That has to be sync (at least without asyncify).

Reply:
Can we at least add a TODO to call more directly into the C API rather than depending on the legacy JS API so we can undo this in the future?

Reply:
Added a todo now. Yes, it would be good to remove that.